projects/npm-tools/packages/npm-scripts/src/utils/createTempFile.js

```diff
@@ -9,8 +9,6 @@ const path = require('path');
 const SignalHandler = require('../utils/SignalHandler');
 const getMergedConfig = require('./getMergedConfig');
 
-const BUILD_CONFIG = getMergedConfig('npmscripts', 'build');
```

Comment:
See the commit description. I believe this has become an anti-pattern in this project. It was not the case before, but I had to refactor the JS Toolkit long ago for the very same reason.

The fact is that we cannot rely on who's calling the functions of this module and, by getting the configuration in advance we are effectively caching it and making the assumption that the functions in the module will always be invoked for the very same project. This is -at the very least- false in the unit test, and can be generally false as soon as we use the tools for more than one project in the same process image.

Reply:
Of course, it would be safer to pass the config as parameter instead of treating it as a global, but for now, using `chdir` should be enough. In fact, it is so widely assume that process.cwd() is the project directory that it is quite difficult to maintain code that works without that assumption.

Reply:
> See the commit description.

Not sure which one you mean exactly, but ok. 😁

Reply:
[This one](https://github.com/liferay/liferay-frontend-projects/pull/436/commits/c0aa550c592da8611ddf8b746df115fb3101e1b2). I referred to the commit containing this change, but it seems that  GH offers an easy way to navigate to it from here...